### DISCUSSION TITLE
OSIC1P - When using cputs() or cprintf() and text needs to scroll the whole screen by 1 line, program goes into infinite loop displaying garbage

### DISCUSSION THREAD

**Discussion by amkissel** [cc65/cc65#2574] [Q&A]
This is for a stock OSI C1P using the standard 24x24 screen, in either the WinOSI emulator or actual hardware.
- Initialize the screen with clrscr()
- Output text to the screen with cputs() or cprintf() functions
- The screen scrolls normally up to the last line on the screen.
- If I perform another cputs() or cprintf(), such that the whole of the screen needs to shift up by 1 line, the program crashes out and just starts to dump garbage to the screen

I must be missing something here.
Has anyone else run into this issue? Am I suppose to scroll the screen in my 'C' code? That doesn't make sense.
Is this a bug?

Looking for any assistance.

Thanks,
Andy

**Comment by mrdudz** [comment]
conio does generally not support scrolling the screen (thats one reason for why it can be small and fast)
